lib/persisted-model.js

```diff
@@ -1527,6 +1546,107 @@ module.exports = function(registry) {
     });
   };
 
+  /**
+   * Subscribe to changes of this model.
+   *
+   * @param {Object} options
+   * @param {Object} options.where Only changes to models matching this where filter will be included in the `ChangeStream`.
+   * @callback {Function} callback
+   * @param {Error} err
+   * @param {ChangeStream} changes
+   */
+
+  PersistedModel.subscribe = function(options, cb) {
+    if(typeof options === 'function') {
+      cb = options;
+      options = undefined;
+    }
+
+    var idName = this.getIdName();
+    var Model = this;
+    var changes = new PassThrough({objectMode: true});
+    var writeable = true;
+    var ref = {changes: changes};
+
+    changes.destroy = function() {
+      changes.removeAllListeners('error');
+      changes.removeAllListeners('end');
+      writeable = false;
+      delete ref.changes;
+    };
+
+    changes.on('error', function() {
+      writeable = false;
+    });
+    changes.on('end', function() {
+      writeable = false;
+    });
+
+    process.nextTick(function() {
+      cb(null, changes);
+    });
+
+    Model.observe('before save', createChangeHandler('save', true));
+    Model.observe('before delete', createChangeHandler('delete', true));
+    Model.observe('after save', createChangeHandler('save'));
+    Model.observe('after delete', createChangeHandler('delete'));
+
+    function createChangeHandler(type, optimistic) {
+      return function(ctx, next) {
+        // note: this leaks the ref and writeable variables
+        // but not the entire change stream (with its huge buffer)
```

Comment:
> thus if A holds a reference to REF and REF holds reference to CHANGES

How is it holding reference after `delete ref.changes`? I am aware that A will hold a reference via a closure variable to REF since the hook code is still referencing / able to invoke A (the hook function).